### DISCUSSION TITLE
Do grule engine is thread safe

### DISCUSSION THREAD

**Discussion by ManouchehrRasoulli** [hyperjumptech/grule-rule-engine#193] [Q&A] [answered]
Hi every body i was using this engine in our code base and i was thinking about engine thread safety, do grule engine is thread safe ?

**Comment by newm4n** [comment, marked as answer]
@ManouchehrRasoulli 

Yes, you can see how grule run engines in goroutine concurrently.
https://github.com/hyperjumptech/grule-rule-engine/blob/master/examples/Concurrency_test.go

As long as the context (and the data there in) are not shared, or thread safe.
